File: tournaments/runLengthEncoding/runLengthEncoding.js
function runLengthEncoding(inputString) {

    var repeatLength = 1,
        answer = [];
    for (var i = 1; i < inputString.length; i++) {
        if (inputString[i] !== inputString[i - 1]) {
            answer.push(repeatLength);
            answer.push(inputString[i - 1]);
            repeatLength = 1;
        }
        else {
            repeatLength++;
        }
    }
    answer.push(repeatLength);
    answer.push(inputString[inputString.length - 1]);

    return answer.join('');
}
    

File: tournaments/runLengthEncoding/runLengthEncoding.cpp
std::string runLengthEncoding(std::string inputString) {

  int repeatLength = 1;
  std::string answer = "";
  for (int i = 1; i < inputString.length(); i++) {
    if (inputString[i] != inputString[i - 1]) {
      answer += std::to_string(repeatLength);
      answer += inputString[i - 1];
      repeatLength = 1;
    }
    else {
      repeatLength++;
    }
  }
  answer += std::to_string(repeatLength);
  answer += inputString[inputString.length() - 1];
  return answer;
}
